tls/extensions/s2n_client_ems.c

```diff
@@ -52,6 +52,16 @@ static int s2n_client_ems_recv(struct s2n_connection *conn, struct s2n_stuffer *
     return S2N_SUCCESS;
 }
 
+/**
+ *= https://www.rfc-editor.org/rfc/rfc7627#section-5.3
+ *= type=exception
+ *# When offering an abbreviated handshake, the client MUST send the
+ *# "extended_master_secret" extension in its ClientHello.
+ *
+ *  We added an exception here in order to prevent a drop in 
+ *  session resumption rates during deployment. Eventually clients
+ *  will be forced to do a full handshake as sessions expire and pick up EMS at that point.
```

Comment:
> Eventually clients will be forced to do a full handshake as sessions expire and pick up EMS at that point.

Do you mean expire in client cache, so that client does not use it? Because on the server side if session expired and full handshake is performed, EMS won't be picked up as client did not send EMS extension in ClientHello.